src/tools/factory-prov-tool/WeaveTLV.py

```diff
@@ -0,0 +1,362 @@
+#!/usr/bin/env python3
+
+#
+#   Copyright (c) 2019 Google LLC.
+#   All rights reserved.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+#
+#   @file
+#         This file contains definitions for working with data encoded in Weave TLV format.
```

Comment:
NOTE TO REVIEWERS: This file will eventually be made part of the openweave python package once that code is decoupled from the Weave Device Manager.  For the time being, however, I'm placing it here so that the factory provisioning tool can operate without dependencies.